Context/ContextSortAttribute.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Markup\NeedleBundle\Context;
+
+use Markup\NeedleBundle\Attribute\AttributeProviderInterface;
+use Markup\NeedleBundle\Config\ContextConfigurationInterface;
+use Markup\NeedleBundle\Sort\RelevanceSort;
+use Markup\NeedleBundle\Sort\Sort;
+
+/**
+ * A context that uses a configuration.
+ */
+class ContextSortAttribute
```

Comment:
Couldn't this be named better? It's not an attribute - more of a factory function.